src/redux/actions/order.js

```diff
@@ -4,29 +4,116 @@ import web3DaiInfura from 'singletons/web3/dai-infura'
 import web3XdaiInfura from 'singletons/web3/xdai-infura'
 import selector from 'redux/selectors'
 
-const add = ({ itemId, quantity }) => {
+/**
+ * Updates or inserts an item in the order.
+ * NOTE: When orderId is not specified current users shopping cart order id is used.
+ * Otherwise, a new order is created and user is updated with the new shopping cart order id.
+ */
+const upsertItem = ({ orderId, itemId, quantity }) => {
   return async (dispatch, getState, { getFirestore }) => {
     const firestore = getFirestore()
     const state = getState()
     const price = selector.getItemPrice(state, { itemId })
-    const items = {
-      [itemId]: { quantity, price },
+    const shoppingCartOrderId = selector.users.shoppingCartOrderId(state)
+    const currentUserId = selector.users.currentId(state)
```

Comment:
```suggestion
    const currentUserId = firebase.auth().currentUser.uid
```

or after authentication save user info into the redux state and use from there